### ISSUE TITLE
Potential merge of efforts

### ISSUE THREAD

**Issue by levithomason** [hallister/semantic-react#62]
Hey there @hallister @asvetliakov @aphofstede and @Asgator.  I'm the creator of [TechnologyAdvice/stardust](https://github.com/TechnologyAdvice/stardust), a React component library for Semantic UI.

Really like direction you guys are going and some of the patterns at play.  Most notably, not using Semantic's JS.  Our team wanted to go this route but didn't have the resources available at the time.  Having ran into all the issues one might expect trying to _properly_ marry jQuery and React, I'm now exploring what it would take to ditch the jQuery.

I think we're duplicating a lot of effort and it would be great to discuss the possibility of collaborating.  Example, I see you're about to add Input label and icon props in #49.  I settled on the same pattern to solve the same issue, see our [input variations](http://technologyadvice.github.io/stardust/#Input-Variations-InputIconExample) docs.

I share a lot of sentiments expressed in this project, especially this from #5:
>...I hate Semantic's build system with a passion... 

Holy crap, yes, unfortunately :/

### What next?

We're dog fooding Stardust in two production apps at the moment.  We've learned a lot about Semantic UI.  I'm sure you have as well.  It would be great to discuss our experiences and patterns.  Even if we don't merge our work, this discussion alone would undoubtedly be beneficial to the both of us.

Obviously, there would be a lot of opinions at play here that we'd have to figure out if we were to collaborate on a single project.  I'm not entirely sure what all those would be.  Though, I can see from the project that we also have a lot in common.

If you're up for it, I'd like to get on a hangout or [appear.in](https://appear.in/) and discuss it more in depth.  If we felt right about it, I think we'd do well to write out some guidelines on how components should be designed and developed.  Then, we'd be well suited to run full speed.  We could also do more publicizing and gather some more help from the community, something we have a knack for at TA.

Let me know if you'd like to chat.  Cheers!

**EDIT**

Fun fact, we also have 6 humans and 2 robots who've [contributed](https://github.com/TechnologyAdvice/stardust/graphs/contributors) to Stardust.  They would all join any collaboration ;)

**Comment by asvetliakov** [comment]
I'm open for chat, as i know @hallister (the original creator) is very busy for now and working on non-React stuff. I'll be really glad if we could collaborate our work, or at least improve our projects.

I believe the current state of our project is pretty solid, most complex components were implemented (popup and dropdown), this project lacks of documentation, examples, some refactoring, tests and build system (CI), or at least few pre-publish scripts.

So these points which we're moving currently on (last few weeks there were no many changes though, since we're working on our proprietary stuff parallely)

**Comment by hallister** [comment]
As @asvetliakov said I'm a little side-tracked in the world of Ember at work right now, but hopefully I will be shifting more into React a day or two a week at some point. Regardless, I can make some time available to chat as well. 

With @asvetliakov's the project is approaching stability. We still have some components (although minor outside of Sticky maybe) that need implemented and a few that need reworked (search). But the majority of remaining work is absolutely docs/examples/tests.

**Comment by levithomason** [comment]
Awesome.  We've got the docs, examples, tests, build system, and CI/CD down for Stardust and can share anything helpful in that respect.  We also have 3 Semantic UI LESS themes with custom gulp builds, as mentioned that was not fun... Two of them are private.  This might be a good topic to cover as well, though I'm most excited to dive into the components themselves.

Stardust is mostly complete and good at what it does, giving simple to use component APIs.  I'd love to discuss this and compare our approaches.

Is there a particular day/time that works best for you guys?  I have openings scattered throughout the week and weekend.

**Comment by aphofstede** [comment]
Not sure how much I could contribute to a call like that. 
I'm adding examples here, so I know something "should work" before hooking it up to my own stuff. This often results in some bugfixes for the components themselves.
From an architecture point of view (Like discussed in #49) I'd just like some consensus / a choice to be made (and documented) on the paradigm to use for component attributes; otherwise it can become messy and inconsistent. 
Haven't checked out Stardust yet.

**Comment by hallister** [comment]
Typically anytime during business hours in PST works for me. I have a ton of meetings Thursday afternoon of this week, but outside of that I can make time whenever it works for you guys.

**Comment by levithomason** [comment]
>I'd just like some consensus / a choice to be made (and documented) on the paradigm to use for component attributes; otherwise it can become messy and inconsistent. 

Couldn't agree more!

Stardust took a very abstracted approach.  We didn't want to write heavily nested HTML looking markup in our apps.  We also wanted to minimize the ramp up burden on developers to memorize that markup.  So we relied on props for the internal structure of a component.  Will be great to chat about all this.

Here are my openings this week in PST:

1. Wed @ 9:30 AM or after 1 PM (tomorrow)
1. Thr anytime before 5 PM
1. Fri @ 9:30 AM or between 3-5 PM

**Comment by asvetliakov** [comment]
Friday looks good for me

**Comment by levithomason** [comment]
### :date:  Friday March 25th @ 3PM
**https://appear.in/semantic-react**

Rough agenda:

- Demo actual usage of both libraries
- Compare source for some specific components
- Review tests, tooling and builds
- Discuss the ideal component API
- Explore collaboration potential

See you there :)

**Comment by hallister** [comment]
Sounds good!

**Comment by hallister** [comment]
Sorry guys I had a power outage so I'm only able to type this via phone. Continue on with the meeting as I'm sure @asvetliakov can discuss more than I can at the moment.

**Comment by levithomason** [comment]
bummer!  will do, take care.

**Comment by levithomason** [comment]
Wanted to say thanks again for the time and collaboration.  I think it was beneficial for the both of us and I hope to touch base again as we each get closer to our `v1.0`.  I'll close this issue for now, thanks again.

**Comment by asvetliakov** [comment]
Thanks you.
Despite of language difference, i think we had informative discussion.
We decided to continue our projects parallel since we have few difference in high-level and consumer api view :smile:

**Comment by levithomason** [comment]
I agree completely, we had a good exchange of information.  It seems that our tooling and development practices are almost identical.

In our meeting last night, we discussed 4 different approaches to component APIs.  We're still deciding what the final API will look like for Stardust.  I'd like to sketch out a few components with each API style we discussed and do two things:

1. Run some performance benchmarks
1. Have our team give their thoughts on all 4

If the result of this exercise leads us to the same (or very similar) components and API as semantic-react, then I can still see a potential for merging our efforts. We'll keep you posted.
